lib/src/util.dart

```diff
@@ -446,3 +446,14 @@ String stripPrivateGeneratedPrefix(String value) {
       ? value.substring(privateGeneratedPrefix.length)
       : value;
 }
+
+/// Takes an iterable of AstNodes and returns their name in a comma separated string.
+///
+/// Optionally, you can pass a function to [getName] if you want something other
+/// than [AstNode.toSource] to be used to generate the name for the list.
+String commaSeparatedAstNodeNames<T extends AstNode>(Iterable<T> nodeList,
+    {String Function(T node) getName}) {
+  getName ??= (node) => node.toSource();
+
+  return nodeList.map(getName).toString().replaceAll(RegExp(r'[\(\)]'), '');
```

Comment:
Using the specific APIs to get the name would be more robust than using toSource and performing string manipulation on the Iterable toString value.

Instead of (from above):
```dart
final otherInterfacesStr = commaSeparatedAstNodeNames(nodeInterfaces);
```
You can do
```dart
final otherInterfacesStr = nodeInterfaces.map((interface) => interface.name.name).join(', ');
```